### PR TITLE
Map view bugs fixed

### PR DESCRIPTION
**Pull request by miajohansson9**
- Map bugs on first load fixed
- Cleaned up order of functions
- Slide Up view bug fixed
- Slide Up view data loaded from array instead of making a request to the database
- Slide Up view now loads each organization's info

### CODE REVIEW

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 1]
```diff
@@ -698,6 +529,145 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
...
+                    public void onFailure(@NonNull Exception e) {
+                        Log.d("MapDemoActivity", "Error trying to get last GPS location");
+                        e.printStackTrace();
+                    }
+                });
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    @NeedsPermission({Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION})
+    protected void startLocationUpdates() {
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: can you test that it works correctly on an emulator with an API lower than M?

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 2]
```diff
@@ -698,6 +529,145 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
...
+                }
+            }
+        });
+
+        if (map != null) {
+            // Map is ready
+            MapDemoFragmentPermissionsDispatcher.getMyLocationWithPermissionCheck(this);
+            MapDemoFragmentPermissionsDispatcher.startLocationUpdatesWithPermissionCheck(this);
+
+            //map.setOnMapLongClickListener(this);
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: delete this?

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 3]
```diff
@@ -698,6 +529,145 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
...
+            map.setInfoWindowAdapter(new CustomWindowAdapter(getLayoutInflater()));
+
+            map.setMapType(GoogleMap.MAP_TYPE_TERRAIN);
+        } else {
+            Toast.makeText(mContext, "Error - Map was null!!", Toast.LENGTH_SHORT).show();
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: I don't think you should display this to the user. Put it in a Log.e statement instead. If you want to show something to the user, just say "Sorry, something went wrong" in the toast.

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 4]
```diff
@@ -735,4 +705,25 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
             }
         });
     }
+
+    public Location getmCurrentLocation() {
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: getCurrentLocation*

**Review by kellymckinnon — APPROVED** [review]
good job!

**Review by ryan-beck — APPROVED** [review]

**Pull request merged**